front_end/src/components/history/CancellationPrompt.js

```diff
@@ -0,0 +1,105 @@
+import React, { Component } from "react";
+import { withStyles } from "@material-ui/core/styles";
+import { connect } from "react-redux";
+import { cancelReservation } from "../../actions/bookingActions";
+import { Button } from "@material-ui/core/";
+
+import Dialog from "@material-ui/core/Dialog";
+import DialogActions from "@material-ui/core/DialogActions";
+import DialogTitle from "@material-ui/core/DialogTitle";
+import { DialogContent } from "@material-ui/core";
+
+const styles = {};
+class CancellationPrompt extends Component {
+  constructor() {
+    super();
+    this.state = {
+      open: false
+    };
+    this.onCancelClick = this.onCancelClick.bind(this);
+  }
+  handleClickOpen = () => {
+    this.setState({ open: true });
+  };
+  handleClose = () => {
+    this.setState({ open: false });
+  };
+
+  onCancelClick(bookingInfo) {
+    const cancelReservationData = { bookingID: bookingInfo };
+    this.props.cancelReservation(cancelReservationData);
+    this.setState({ open: false });
+    window.location.reload();
+  }
+  render() {
+    const { classes, width, checkInTime, id } = this.props;
+    return (
+      <div>
+        <Button
+          onClick={this.handleClickOpen}
+          variant="outlined"
+          color="secondary"
+          className={classes.button}
+        >
+          CANCEL
+        </Button>
+
+        <Dialog
+          fullWidth={
+            width === "md" || width === "lg" || width === "sm" ? true : false
+          }
+          fullScreen={width === "xs" ? true : false}
+          open={this.state.open}
+          onClose={this.handleClose}
+        >
+          <DialogTitle>Are you sure you want to cancel?</DialogTitle>
+          {new Date(checkInTime).getTime() - new Date().getTime() >
+          172800000 ? (
+            <DialogContent>
+              <DialogContent>
+                Full refund provided for this cancellation.
```

Comment:
can we tell the user which hotel he is canceling?


Are you sure you want to cancel the following reservation?
Marriot Hotel Miami
Check-in: 4/4/2019
check-out: 5/5/2019

This booking qualifies for a full refund if canceled today.

Reply:
fixed!